provisioning/miniprov/hare_mp/main.py

```diff
@@ -35,6 +35,7 @@
 
 from hare_mp.cdf import CdfGenerator
 from hare_mp.store import ConfStoreProvider
+from hare_mp.types import MissingKey
```

Comment:
How about prefixing or suffixing with Exception or Error?
I believe there's already a KeyError exception for json, no? cannot we use that, why do we need to have our own exception in this case?

Reply:
@mssawant Because KeyError is probably not the only "symptom"  that the key is not found in ConfStore. For instance, ConfStore can contain an array with incomplete set of indices or something like that. Since we don't want to assume so many different exceptions at the top level of main.py, I suggested to create our own exception. So if we see exception MissingKey, it means that ConfStore doesn't match our expectations. No matter how the underlying layer came to that decision.

Reply:
>For instance, ConfStore can contain an array with incomplete set of indices or something like that.

Okay, so may be we got server_nodes count as 3 but only 2 of them are described. I somehow still think its a KeyError. But its okay, if we think that a separate exception class is more elaborate then let's use a separate exception class. I am fine with it.